shared/src/business/useCases/document/batchDownloadDocketEntriesInteractor.ts

```diff
@@ -0,0 +1,204 @@
+    throw new UnauthorizedError('Unauthorized');
+  }
+
+  const caseToBatch = await applicationContext
+    .getPersistenceGateway()
+    .getCaseByDocketNumber({
+      applicationContext,
+      docketNumber,
+    });
+
+  if (!caseToBatch) {
+    throw new NotFoundError(`Case: ${docketNumber} was not found.`);
+  }
+
+  const s3Ids: string[] = [];
+  const fileNames: string[] = [];
+  const extraFileNames: string[] = [];
+  const extraFiles: string[] = [];
+
+  const { caseCaption, isSealed: isCaseSealed } = caseToBatch;
+  const caseTitle = Case.getCaseTitle(caseCaption);
+  const caseFolder = `${docketNumber}, ${caseTitle}`;
+  const zipName = `${caseFolder}.zip`;
+
+  const caseEntity = new Case(caseToBatch, { applicationContext });
+
+  const documentsToProcess = documentsSelectedForDownload.map(docketEntryId =>
+    caseEntity.getDocketEntryById({
+      docketEntryId,
+    }),
+  );
+
+  documentsToProcess.forEach(docInfo => {
+    const { docketEntryId, documentTitle, filingDate, index } = docInfo;
+    const filename = generateValidDocketEntryFilename({
+      documentTitle,
+      filingDate,
+      index,
+    });
+
+    const fileDirectory =
+      isCaseSealed || docInfo.isSealed ? `${caseFolder}/sealed` : caseFolder;
+
+    const pdfTitle = docInfo.isStricken ? `STRICKEN_${filename}` : filename;
+
+    s3Ids.push(docketEntryId);
+    fileNames.push(`${fileDirectory}/${pdfTitle}`);
+  });
+
+  if (printableDocketRecordFileId) {
+    const pdfTitle = `${caseFolder}/${'0_Docket_Record.pdf'}`;
+
+    const printableDocketRecordPdf: any = await applicationContext
+      .getPersistenceGateway()
+      .getDocument({
+        applicationContext,
+        key: printableDocketRecordFileId,
+        useTempBucket: true,
+      });
+    extraFiles.push(printableDocketRecordPdf);
+    extraFileNames.push(pdfTitle);
+  }
+
+  const onEntry = entryData => {
+    applicationContext.getNotificationGateway().sendNotificationToUser({
+      applicationContext,
+      clientConnectionId,
+      message: {
+        action: 'batch_download_entry',
+        ...entryData,
+        numberOfDocketRecordsToGenerate: 0,
+        numberOfFilesToBatch: s3Ids.length + extraFiles.length,
+      },
+      userId: user.userId,
+    });
+  };
+
+  const onError = error => {
+    applicationContext.logger.error('Archive Error', { error });
+    applicationContext.getNotificationGateway().sendNotificationToUser({
+      applicationContext,
+      clientConnectionId,
+      message: {
+        action: 'batch_download_error',
+        error,
+      },
+      userId: user.userId,
+    });
+  };
+
+  const onProgress = progressData => {
+    applicationContext.getNotificationGateway().sendNotificationToUser({
+      applicationContext,
+      clientConnectionId,
+      message: {
+        action: 'batch_download_progress',
+        ...progressData,
+        numberOfDocketRecordsToGenerate: 0,
```

Comment:
is this number always set to zero?

Reply:
The batch download operation is written to download cases and documents associated to those cases. For this story, we are only downloading documents on a case, hence, 0 for case downloads.